presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/TypeUtils.java

```diff
@@ -0,0 +1,261 @@
+            .put(DATE, "DATE")
+            .put(TIME, "TIME")
+            .put(TIME_WITH_TIME_ZONE, "TIME")
+            .put(TIMESTAMP, "TIMESTAMP")
+            .put(TIMESTAMP_WITH_TIME_ZONE, "TIMESTAMP")
+            .build();
+
+    private TypeUtils()
+    {
+    }
+
+    public static boolean isArrayType(Type type)
+    {
+        return type.getTypeSignature().getBase().equals(StandardTypes.ARRAY);
+    }
+
+    public static boolean isMapType(Type type)
+    {
+        return type.getTypeSignature().getBase().equals(StandardTypes.MAP);
+    }
+
+    public static boolean isRowType(Type type)
+    {
+        return type.getTypeSignature().getBase().equals(StandardTypes.ROW);
+    }
+
+    public static boolean isDateType(Type type)
+    {
+        return type.equals(DATE) ||
+                type.equals(TIME) ||
+                type.equals(TIMESTAMP) ||
+                type.equals(TIMESTAMP_WITH_TIME_ZONE);
+    }
+
+    public static String toSqlType(Type type)
+    {
+        if (type instanceof VarcharType) {
+            if (((VarcharType) type).isUnbounded()) {
+                return "VARCHAR";
+            }
+            return "VARCHAR(" + ((VarcharType) type).getBoundedLength() + ")";
+        }
+        if (type instanceof CharType) {
+            if (((CharType) type).getLength() == CharType.MAX_LENGTH) {
+                return "CHAR";
+            }
+            return "CHAR(" + ((CharType) type).getLength() + ")";
+        }
+
+        String sqlType = SQL_TYPES.get(type);
+        if (sqlType != null) {
+            return sqlType;
+        }
+
+        if (type instanceof DecimalType) {
+            return type.toString().toUpperCase();
+        }
+
+        if (isArrayType(type)) {
+            Type elementType = type.getTypeParameters().get(0);
+            sqlType = toSqlType(elementType);
+            return sqlType + " ARRAY[]";
+        }
+
+        throw new PrestoException(NOT_SUPPORTED, "Unsupported column type: " + type.getTypeSignature());
+    }
+
+    public static Type toPrestoType(int phoenixType, int columnSize, int decimalDigits, int arraySize, int rawTypeId)
+    {
+        switch (phoenixType) {
+            case Types.BIT:
+            case Types.BOOLEAN:
+                return BOOLEAN;
+            case Types.TINYINT:
+                return TINYINT;
+            case Types.SMALLINT:
+                return SMALLINT;
+            case Types.INTEGER:
+                return INTEGER;
+            case Types.BIGINT:
+                return BIGINT;
+            case Types.REAL:
+                return REAL;
+            case Types.FLOAT:
+            case Types.DOUBLE:
+            case Types.NUMERIC:
+                return DOUBLE;
+            case Types.DECIMAL:
+                return DecimalType.createDecimalType(columnSize, decimalDigits);
+            case Types.CHAR:
+            case Types.NCHAR:
+                return createCharType(min(columnSize, CharType.MAX_LENGTH));
```

Comment:
if `columnSize > CharType.MAX_LENGTH`, you need to represent the data as `varchar`. 
Otherwise, what would you do with the data values that are too long to fit?